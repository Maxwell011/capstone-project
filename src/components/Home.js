import React, { useEffect } from "react";
import { fetchPosts, educationPosts, sciencePosts, healthPosts, techPosts } from "../redux/HomeSlice";
import { useSelector, useDispatch } from "react-redux";
import Posts from './Posts';


const Home = () => {
 const dispatch = useDispatch();
 const  { posts } = useSelector((state) => state.posts)

useEffect(() => {
 dispatch(fetchPosts());
}, [dispatch]);

const handleClick = () => {
  dispatch(fetchPosts());
};

const handleDisplayEducationSearch = () => {
  dispatch(educationPosts());
}

const handleDisplayScienceSearch = () => {
  dispatch(sciencePosts());
}

const handleDisplayHealthSearch = () => {
  dispatch(healthPosts());
}

const handleDisplayTechSearch = () => {
  dispatch(techPosts());
}

     console.log(posts);

  return (
    <div>
      <button onClick={handleClick}>All</button>
      <button onClick={handleDisplayEducationSearch}>Education</button>
      <button onClick={handleDisplayScienceSearch}>Science</button>
      <button onClick={handleDisplayHealthSearch}>Health</button>
      <button onClick={handleDisplayTechSearch}>Technology</button>
      {posts?.data?.map((post) => (
        <Posts key={post.id} post={post}></Posts>
      ))}
    </div>
  );
};

export default Home;

