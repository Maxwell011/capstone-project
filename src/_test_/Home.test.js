import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
