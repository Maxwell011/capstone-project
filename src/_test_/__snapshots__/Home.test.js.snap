// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="carousel carousel slide"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="carousel-indicators"
    >
      <button
        aria-current={true}
        aria-label="Slide 1"
        className="active"
        data-bs-target=""
        onClick={[Function]}
        type="button"
      />
      <button
        aria-current={false}
        aria-label="Slide 2"
        data-bs-target=""
        onClick={[Function]}
        type="button"
      />
      <button
        aria-current={false}
        aria-label="Slide 3"
        data-bs-target=""
        onClick={[Function]}
        type="button"
      />
      <button
        aria-current={false}
        aria-label="Slide 4"
        data-bs-target=""
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="carousel-inner"
    >
      <div
        className="active carousel-item"
        interval={1000}
      >
        <img
          alt="First slide"
          className="d-block w-100 "
          src="BURJ KHALIFA.jpg"
        />
        <div
          className="carousel-caption"
        >
          <h3
            className="burj-head"
          >
            BURJ KHALIFA
          </h3>
          <p
            className="burj-khalifa"
          >
            The Burj Khalifa is a mixed-use skyscraper located in Dubai, U.A.E. It is the world’s tallest building, according to all three of the main criteria by which such buildings are judged. The Burj Khalifa was officially named to honour the president of the neighbouring emirate of Abu Dhabi, Sheikh Khalīfah ibn Zāyid Āl Nahyān.
            <button
              className="btn btn-tag btn-rounded btn btn-outline-secondary"
              disabled={false}
              type="button"
            >
              Read More
            </button>
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
        interval={500}
      >
        <img
          alt="Second slide"
          className="d-block w-100"
          src="BURJ KHALIFA2.jpg"
        />
        <div
          className="carousel-caption"
        >
          <h3>
            BURJ KHALIFA
          </h3>
          <p>
            The Burj Khalifa is a mixed-use skyscraper located in Dubai, U.A.E. It is the world’s tallest building, according to all three of the main criteria by which such buildings are judged. The Burj Khalifa was officially named to honour the president of the neighbouring emirate of Abu Dhabi, Sheikh Khalīfah ibn Zāyid Āl Nahyān.
            <button
              className="btn btn-tag btn-rounded btn btn-outline-secondary"
              disabled={false}
              type="button"
            >
              Read More
            </button>
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Third slide"
          className="d-block w-100"
          src="Workplace.jpg"
        />
        <div
          className="carousel-caption"
        >
          <h3>
            Work Space
          </h3>
          <p>
            This is the new modern workspace and this is how you can set up your own work space as a developer or a designer.
            <button
              className="btn btn-tag btn-rounded btn btn-outline-secondary"
              disabled={false}
              type="button"
            >
              Read More
            </button>
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Fourth slide"
          className="d-block w-100"
          src="Efeirtower.jpg"
        />
        <div
          className="carousel-caption"
        >
          <h3
            className="eiffer-head"
          >
            The Eiffel Tower
          </h3>
          <p
            className="eiffel-tower"
          >
            The Eiffel Tower is the most visited monument with an entrance fee in the world: 6.91 million people ascended it in 2015. It was designated a monument history in 1964, and was named part of a UNESCO World Heritage Site ("Paris, Banks of the Seine") in 1991.
            <button
              className="btn btn-tag btn-rounded btn btn-outline-secondary"
              disabled={false}
              type="button"
            >
              Read More
            </button>
          </p>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="visually-hidden"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="visually-hidden"
      >
        Next
      </span>
    </a>
  </div>
  <div
    className="categories"
  >
    <button
      className="btn btn-tag btn-rounded btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <button
      className="btn btn-tag btn-rounded btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Education
    </button>
    <button
      className="btn btn-tag btn-rounded btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Science
    </button>
    <button
      className="btn btn-tag btn-rounded btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Health
    </button>
    <button
      className="btn btn-tag btn-rounded btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Technology
    </button>
  </div>
  <div
    className="home-container"
  />
</div>
`;
